terraform/mysql/bind/variables.tf

```diff
@@ -16,8 +16,16 @@ variable "db_name" { type = string }
 variable "hostname" { type = string }
 variable "admin_username" { type = string }
 variable "admin_password" { type = string }
-variable "use_tls" { type = bool }
+variable "require_ssl" { type = bool }
 
 locals {
   port = 3306
+
+  # The following values are allowed:
+  # "DISABLED" - Establish unencrypted connections;
+  # "PREFERRED" - Establish encrypted connections if the server enabled them, otherwise fall back to unencrypted connections;
+  # "REQUIRED" - Establish secure connections if the server enabled them, fail otherwise;
+  # "VERIFY_CA" - Like "REQUIRED" but additionally verify the server TLS certificate against the configured Certificate Authority (CA) certificates;
+  # "VERIFY_IDENTITY" - Like "VERIFY_CA", but additionally verify that the server certificate matches the host to which the connection is attempted.
+  sslMode = var.require_ssl ? "REQUIRED" : "PREFERRED"
```

Comment:
Does it make sense to use `require_ssl` in this binding? This is giving `require_ssl` two meanings:
1. It does not allow the DB to be accessed via non-SSL connections - the intended story 
2. It  defines how the user should connect - which can be several ways and still be over SSL.

If we want to keep both behaviours does it make sense to expose them separately?